### PR TITLE
Fixes #1843 Launch and kill a worker or master

### LINKED ISSUE #1843: Launch and kill a worker or master
Privileged person should have the ability add a new worker (or a master) or kill a running worker (or a master) via REST api.

### PR DESCRIPTION
**Pull request by kkasravi**

### CODE REVIEW

**Comment by manuzhang** [comment]
```
[error] /home/travis/build/gearpump/gearpump/services/jvm/src/test/scala/io/gearpump/services/StaticServiceSpec.scala:34: not enough arguments for constructor StaticService: (system: akka.actor.ActorSystem, supervisorPath: String)io.gearpump.services.StaticService.
[error] Unspecified value parameter supervisorPath.
[error]   def route = new StaticService(system).route
[error] 
```

**Comment by clockfly** [comment]
+1 for the code after getting the UT fixed.

**Comment by clockfly** [comment]
Compile failed.

**Comment by clockfly** [comment]
```
[error] (gearpump-experiments-yarn/test:compileIncremental) Compilation failed
[error] (gearpump-streaming/test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 426 s, completed Jan 20, 2016 8:13:12 AM
```

**Comment by clockfly** [comment]
```
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/appmaster/YarnAppMasterSpec.scala:163: not found: value AddWorker
[error] Error occurred in an application involving default arguments.
[error]     appMaster.tell(AddWorker(newWorkerCount), client.ref)
[error]                    ^
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/appmaster/YarnAppMasterSpec.scala:164: not found: type CommandResult
[error] Error occurred in an application involving default arguments.
[error]     client.expectMsgType[CommandResult]
[error]                          ^
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/appmaster/YarnAppMasterSpec.scala:181: not found: value RemoveWorker
[error] Error occurred in an application involving default arguments.
[error]     appMaster.tell(RemoveWorker(workerContainerId.toString), client.ref)
[error]                    ^
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/appmaster/YarnAppMasterSpec.scala:182: not found: type CommandResult
[error] Error occurred in an application involving default arguments.
[error]     client.expectMsgType[CommandResult]
[error]                          ^
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/appmaster/YarnAppMasterSpec.scala:193: not found: type CommandResult
[error]     client.expectMsgType[CommandResult]
[error]                          ^
[INFO] [01/20/2016 08:11:48.500] [SessionTrackerImpl] SessionTrackerImpl exited loop!
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/client/ManageClusterSpec.scala:26: value RemoveWorker is not a member of object io.gearpump.experiments.yarn.appmaster.YarnAppMaster
[error] import io.gearpump.experiments.yarn.appmaster.YarnAppMaster.{Kill, ClusterInfo, QueryClusterInfo, Version, QueryVersion, RemoveWorker, CommandResult, AddWorker, ActiveConfig, GetActiveConfig}
[error]        ^
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/client/ManageClusterSpec.scala:71: not found: value AddWorker
[error]     appMaster.expectMsg(AddWorker(1))
[error]                         ^
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/client/ManageClusterSpec.scala:84: not found: value RemoveWorker
[error]     appMaster.expectMsg(RemoveWorker("1"))
[error]                         ^
[INFO] [01/20/2016 08:11:48.983] [Slf4jLogger] Slf4jLogger started
[error] /home/travis/build/gearpump/gearpump/experiments/yarn/src/test/scala/io/gearpump/experiments/yarn/client/ManageClusterSpec.scala:128: not found: value CommandResult
[error]     appMaster.reply(CommandResult(true))
[error]                     ^
[error] 10 errors found
```

File: core/src/main/scala/io/gearpump/util/ActorUtil.scala [hunk 1]
```diff
@@ -113,6 +113,11 @@ object ActorUtil {
     appmaster.flatMap(askActor[T](_, msg))
   }
 
+  def askSupervisor[T](supervisor: ActorRef, count: Int, msg: Any)(implicit ex: ExecutionContext): Future[T] = {
```
Comment (clockfly) [inline core/src/main/scala/io/gearpump/util/ActorUtil.scala]: Should we make this private to SupervisorService?

**Pull request merged**